src/screens/SosContactEdit/index.js

```diff
@@ -0,0 +1,152 @@
+import React, { useContext, useState, useEffect } from 'react';
+import {
+  View,
+  Text,
+  StyleSheet,
+  TextInput,
+  TouchableOpacity,
+} from 'react-native';
+import { FontAwesomeIcon } from '@fortawesome/react-native-fontawesome';
+import { faTrash } from '@fortawesome/free-solid-svg-icons';
+import { useIsFocused } from '@react-navigation/native';
+import appApiClient from '../../api/appApiClient';
+
+import { Context as AuthContext } from '../../state/AuthContext';
+
+export default function SosContactEdit({ navigation, route }) {
+  const [contact, setContact] = useState({});
+  const { state } = useContext(AuthContext);
+  const isFocused = useIsFocused();
+
+  useEffect(() => {
+    getContact();
+  }, [isFocused]);
+
+  // find the selected contact and add it in contact state
+  const getContact = async () => {
+    try {
+      const response = await appApiClient.get(
+        `/users/${state.username}/contacts/`,
+      );
+
+      const foundContact = response.data.contacts.find(
+        (item) => item._id === route.params.id,
+      );
+      setContact(foundContact);
+    } catch (error) {
+      console.error(error);
+    }
+  };
+
+  const handleNameChange = (name) => {
+    setContact({ ...contact, name });
+  };
+
+  const handleNumberChange = (phone) => {
+    setContact({ ...contact, phone });
+  };
+
+  const handleMessageChange = (message) => {
+    setContact({ ...contact, message });
+  };
+
+  const saveEdit = () => {
+    const data = {
+      name: contact.name,
+      phone: contact.phone,
+      message: contact.message,
+    };
+    appApiClient
+      .patch(`/users/${state.username}/contacts/${route.params.id}`, data)
+      .then((response) => {
+        alert(response.data);
+      })
+      .catch((e) => {
+        alert(e);
+      });
+  };
+
+  const handleRemove = async (id) => {
+    appApiClient
+      .delete(`/users/${state.username}/contacts`, {
+        params: { id },
+      })
+      .then((response) => {
+        console.log(response.data);
+      })
```

Comment:
console log response? may be better to do something like alerting that "contact was deleted"?

Reply:
response.data is a message that says contact was successfully deleted, ill change to alert